api/metrics-api-common/src/main/java/org/hawkular/metrics/api/jaxrs/util/OriginValidation.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2014-2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.metrics.api.jaxrs.util;
+
+import java.net.URI;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * @author Stefan Negrea
+ */
+public final class OriginValidation {
+
```

Comment:
Implementation could be more object oriented by turning this class into a CDI bean:
- you can inject CORS config right here instead of `CorsFilter`
- `parseAllowedCorsOrigins` becomes a `@PostConstruct` method

This would clearly separate concerns: origin validation here, header manipulation in `CorsFilter`.